src/main/java/htsjdk/samtools/liftover/LiftOver.java

```diff
@@ -84,9 +85,16 @@ public long getFailedIntervalsBelowThreshold() {
     /**
      * Load UCSC chain file in order to lift over Intervals.
      */
-    public LiftOver(File chainFile) {
-        IOUtil.assertFileIsReadable(chainFile);
-        chains = Chain.loadChains(chainFile);
+    public LiftOver(File chainFile){
+        this(toInputStream(chainFile), chainFile.toString());
```

Comment:
It seems like java constructor constraints have forced some slightly awkward behavior here.

I would probably restructure like this in order to avoid some awkwardness.

```
 /**
     * Load UCSC chain file in order to lift over Intervals.
     */
    public LiftOver(File chainFile){
        this(Chain.loadChains(chainFile));
    }

    /**
     * Load UCSC chain file in order to lift over Intervals.
     */
    public LiftOver(InputStream chainFileInputStream, String sourceName) {
        this(Chain.loadChains(new BufferedLineReader(chainFileInputStream), sourceName));
    }

    private LiftOver(OverlapDetector<Chain> chains) {
        this.chains = chains;
        for (final Chain chain : this.chains.getAll()) {
            final String from = chain.fromSequenceName;
            final String to   = chain.toSequenceName;
            final Set<String> names;
            if (contigMap.containsKey(from)) {
                names = contigMap.get(from);
            }
            else {
                names = new HashSet<>();
                contigMap.put(from, names);
            }
            names.add(to);
        }
    }
```

That way there's no question about who's responsibility it is to close streams.  (sorry to harp on this particular issue, it's just something that's come up a bunch recently so it's in my mind...)